### PR TITLE
[merge] Fix/unit tests

### PR DESCRIPTION
**Pull request by MoKob**
Addressing https://github.com/Project-OSRM/osrm-backend/issues/2218

 - [x] find reason that breaks cucumber tests
 - [x] fix server-tests
 - [x] @daniel-j-h :+1:

### CODE REVIEW

File: unit_tests/engine/base64.cpp [hunk 1]
```diff
@@ -16,7 +16,6 @@ BOOST_AUTO_TEST_CASE(rfc4648_test_vectors)
 {
     using namespace osrm::engine;
 
-    BOOST_CHECK_EQUAL(encodeBase64(""), "");
```
Comment (MoKob) [inline unit_tests/engine/base64.cpp]: this is asserted as invalid input

**Comment by TheMarex** [comment]
The remaining cucumber test was :construction_worker: :trollface: Restarting it worked. This works now. :green_heart: Great work!

File: include/util/static_rtree.hpp [hunk 2]
```diff
@@ -234,21 +233,20 @@ class StaticRTree
...
-                                      const std::uint32_t old_id = current_tree_node.children[j];
-                                      const std::uint32_t new_id = search_tree_size - old_id - 1;
-                                      current_tree_node.children[j] = new_id;
-                                  }
-                              }
-                          });
+        tbb::parallel_for(
+            tbb::blocked_range<std::uint32_t>(0, search_tree_size),
+            [this, &search_tree_size](const tbb::blocked_range<std::uint32_t> &range) {
+                for (std::uint32_t i = range.begin(), end = range.end(); i != end; ++i)
```
Comment (TheMarex) [inline include/util/static_rtree.hpp]: All these lambda related style changes make me suspect there is an option in our stylefile missing that is used in `3.8`.

**Comment by TheMarex** [comment]
Looks good to me. @MoKob is this merge ready?

File: include/util/static_rtree.hpp (hunk 2 above)
Comment (daniel-j-h) [inline include/util/static_rtree.hpp]: So let's update the style file. There is at least `SortIncludes` (on by default) that we can specify in addition; `-dump-config` and go from there.

**Comment by MoKob** [comment]
It is, as long as @daniel-j-h does not have further comments. It requires https://github.com/Project-OSRM/osrm-backend/pull/2133 as a base though to have roundabout/rotaries tests working

File: include/server/api/base_parameters_grammar.hpp [hunk 3]
```diff
@@ -72,7 +72,7 @@ struct BaseParametersGrammar : boost::spirit::qi::grammar<std::string::iterator>
 
         alpha_numeral = +qi::char_("a-zA-Z0-9");
         polyline_chars = qi::char_("a-zA-Z0-9_.--[]{}@?|\\%~`^");
-        base64_char = qi::char_("a-zA-Z0-9--_");
+        base64_char = qi::char_("a-zA-Z0-9--_=");
```
Comment (daniel-j-h) [inline include/server/api/base_parameters_grammar.hpp]: Why `--`? /cc @TheMarex

File: include/util/static_rtree.hpp [hunk 4]
```diff
@@ -473,22 +468,22 @@ class StaticRTree
         // current object represents a block on disk
         for (const auto i : irange(0u, current_leaf_node.object_count))
         {
-            auto &current_edge = current_leaf_node.objects[i];
-            auto projected_u =
+            const auto &current_edge = current_leaf_node.objects[i];
+            const auto projected_u =
                 coordinate_calculation::mercator::fromWGS84((*m_coordinate_list)[current_edge.u]);
-            auto projected_v =
+            const auto projected_v =
```
Comment (daniel-j-h) [inline include/util/static_rtree.hpp]: :+1: on const!

File: src/util/coordinate_calculation.cpp [hunk 5]
```diff
@@ -104,10 +104,11 @@ std::pair<double, FloatCoordinate> projectPointOnSegment(const FloatCoordinate &
     {
         clamped_ratio = 0.;
     }
+
     return {clamped_ratio,
             {
-                source.lon + slope_vector.lon * FloatLongitude(clamped_ratio),
-                source.lat + slope_vector.lat * FloatLatitude(clamped_ratio),
+                FloatLongitude(1.0-clamped_ratio) * source.lon + target.lon * FloatLongitude(clamped_ratio),
+                FloatLatitude(1.0-clamped_ratio) * source.lat + target.lat * FloatLatitude(clamped_ratio),
```
Comment (daniel-j-h) [inline src/util/coordinate_calculation.cpp]: Format

File: unit_tests/library/route.cpp [hunk 6]
```diff
@@ -32,31 +32,48 @@ BOOST_AUTO_TEST_CASE(test_route_same_coordinates_fixture)
     const auto rc = osrm.Route(params, result);
     BOOST_CHECK(rc == Status::Ok);
 
+    //unset snapping dependent hint
+    for( auto & itr : result.values["waypoints"].get<json::Array>().values )
```
Comment (daniel-j-h) [inline unit_tests/library/route.cpp]: Format

File: unit_tests/util/static_rtree.cpp [hunk 7]
```diff
@@ -1,28 +1,28 @@
...
+#include "util/coordinate_calculation.hpp"
+#include "util/exception.hpp"
+#include "util/rectangle.hpp"
 #include "util/static_rtree.hpp"
+#include "util/typedefs.hpp"
 
 #include "mocks/mock_datafacade.hpp"
 
-#include <boost/test/unit_test.hpp>
+#include <boost/functional/hash.hpp>
```
Comment (daniel-j-h) [inline unit_tests/util/static_rtree.cpp]: Out of interest, why is this needed?

File: include/server/api/base_parameters_grammar.hpp (hunk 3 above)
Comment (TheMarex) [inline include/server/api/base_parameters_grammar.hpp]: Because in boost spirit `-` doesn't mean what you think it means. This is escaping for `-`.

File: unit_tests/util/static_rtree.cpp (hunk 7 above)
Comment (MoKob) [inline unit_tests/util/static_rtree.cpp]: no idea. its simply a sorting by format

**Comment by TheMarex** [comment]
:tada: merged
